protecode/scanning_util.py

```diff
@@ -231,40 +227,45 @@ def _metadata(
         metadata.update(self._component_metadata(component=component, omit_version=omit_version))
         return metadata
 
-    def upload_container_image_group(
+    def process_component_resources(
         self,
-        resource_group,
+        component_resources: typing.Iterable[cnudie.util.ComponentResource],
     ) -> typing.Iterable[UploadResult]:
 
-        logger.info(
-            f'Processing resource group for component {resource_group[0].component.name} and '
-            f'resource {resource_group[0].resource.name} with {len(resource_group)}'
-        )
-
         # depending on upload-mode, determine an upload-action for each related image
-        # - images to upload
+        # - resources to upload
         # - protecode-apps to remove
         # - triages to import
-        images_to_upload = list()
-        protecode_apps_to_remove = set()
-        protecode_apps_to_consider = list() # consider to rescan; return results
+        product_id_to_resource: dict[str, cnudie.util.ComponentResource] = dict()
+        protecode_products_to_consider = list() # consider to rescan; return results
+        protecode_products_to_remove = set()
+        component_resources_to_upload = list()
         triages_to_import = set()
-        product_id_to_resource = dict()
+
+        # HACK since the component and resource name are the same for all elements 
+        # (only the component and resource version differ) we can use the names for all elements
+        first_component_resource = next(component_resources.__iter__())
```

Comment:
according to typehint, `component_resources` might be any iterable (including e.g. a generator - or, more generically speaking - an iterable that will always return the same iterator object). therfore, using this pattern (and declaring any iterable will be okay) will lead to a subtle bug if such a "non-repeatable" iterable is passed-in: if this happens, the first element will be removed from it, and it will not be included later in this function.

therefore, I deem it preferable to either:
- change typehint to `list[T]` and use `component_resources[0]` to get the first element (this will at least fail during runtime if e.g. a generator is passed in)
or:
- convert iterable into a list (or tuple). also in the second case, I would deem using `[0]` to get the first element more idiosyncratic (+ easier to read / shorter)